### PR TITLE
Add rb support using turbine-core

### PR DESCRIPTION
**Pull request by jmar910**
## Description of change

Changes to CLI for Turbine Ruby project. Addresses basic ruby stubbing out, workable run action and draft of deploy action.

Fixes 

- https://github.com/meroxa/cli/issues/522
- https://github.com/meroxa/cli/issues/521

Drafts  

- https://github.com/meroxa/cli/issues/518
- https://github.com/meroxa/cli/issues/523

## Type of change

- [ ]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

### CODE REVIEW

File: cmd/meroxa/root/apps/deploy_test.go [hunk 1]
```diff
@@ -1110,9 +1115,9 @@ func Test_getTurbineCLIFromLanguage(t *testing.T) {
 			wantTurbineCLI: turbinePY.New(d.logger, d.path),
 		},
 		{
-			name:           "when language is python",
-			language:       turbine.Python,
-			wantTurbineCLI: turbinePY.New(d.logger, d.path),
+			name:           "when language is ruby",
+			language:       turbine.Ruby,
```
Comment (anna-cross) [inline cmd/meroxa/root/apps/deploy_test.go:1119]: Not here, py was checked twice, so i just turned second one into ruby check lol

File: cmd/meroxa/turbine/ruby/build.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package turbinerb
+
+import (
+	"context"
+)
+
+// Build has nothing to do for turbine-rb.
+func (t *turbineRbCLI) Build(ctx context.Context, appName string, platform bool) (string, error) {
+	return "", nil
+}
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/build.go:10]: Isn't this important for building a function? So it is something that we should be doing here

File: cmd/meroxa/turbine/ruby/deploy.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+			"unable to determine if the Meroxa Application at %s has a Process; %s",
+			t.appPath,
+			string(output))
+		return false, err
+	}
+	return strconv.ParseBool(isNeeded)
+}
+
+func (t *turbineRbCLI) Deploy(ctx context.Context, imageName, appName, gitSha, specVersion, accountUUID string) (string, error) {
+	var (
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/deploy.go:40]: you'll need to start up the grpc server here in `platform` mode and then get deployment spec from there

File: cmd/meroxa/turbine/ruby/deploy.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+	return deploymentSpec, err
+}
+
+// GetResources asks turbine for a list of resources used by the given app.
+func (t *turbineRbCLI) GetResources(ctx context.Context, appName string) ([]utils.ApplicationResource, error) {
+	var resources []utils.ApplicationResource
+
+	cmd := exec.CommandContext(ctx, "turbine_client", "listresources", t.appPath)
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/deploy.go:85]: This information needs to be retrieved from the gRPC server, and not from the ruby library

File: cmd/meroxa/turbine/ruby/init.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package turbinerb
+
+import (
+	"context"
+
+	utils "github.com/meroxa/cli/cmd/meroxa/turbine"
+)
+
+func (t *turbineRbCLI) Init(ctx context.Context, name string) error {
+	cmd := utils.RunTurbineJS(ctx, "generate", name, t.appPath)
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/init.go:10]: Why is this running JS? 
Also, we're not implementing this command until Milestone 2

File: cmd/meroxa/turbine/ruby/upgrade.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package turbinerb
+
+import (
+	"os/exec"
+)
+
+// Upgrade fetches the latest Meroxa dependencies.
+func (t *turbineRbCLI) Upgrade(vendor bool) error {
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/upgrade.go:8]: We're not implementing this command, so this should be removed

File: cmd/meroxa/turbine/ruby/build.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package turbinerb
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/ruby/build.go:1]: The whole turbine ruby package still needs testing

**Review by dianadoherty — COMMENTED** [review]
I think this PR needs to be split up, and tested more, each feature at a time. First run, then deploy, then run with functions, then deploy with functions

**Pull request closed without merging**